templates/ovndbcluster/bin/cleanup.sh

```diff
@@ -20,6 +20,22 @@ DB_TYPE="{{ .DB_TYPE }}"
 if [[ "${DB_TYPE}" == "sb" ]]; then
     DB_NAME="OVN_Southbound"
 fi
+
+# TODO: this logic is dedicated for non-leaders. The assumption that -0 is
+# always the leader is invalid and a bug that should be fixed.
```

Comment:
just out of curiosity: do we have jira reported for that bug already?

Reply:
I am not aware of it. We should report one.

Reply:
+1 to have a bug if we need any optimization here.

Reasoning wise this was done to never leave the only node in the cluster and drop the database, considering scale down replicas=0 case and all pod deleted. Once we scale back to >=1 we have same cluster started instead of new bootstrap(new cid/sid)

Reply:
I think you are right, indeed applying the same logic for all pods makes it problematic, because then not a single member is left to hold onto the database file and the cluster leadership.

For the case of scaling down to zero, retaining the database may still make sense for cid/sid.

I will update the comments to explain the actual reason for this. I misinterpreted the intent of the -0 name check as leadership check, but it's just a check for a "single pod in a cluster, even if it's not a leader."